e2e/testcases/otel_collector_test.go

```diff
@@ -0,0 +1,215 @@
+	nt.T.Cleanup(func() {
+		nt.MustKubectl("delete", "-f", "../testdata/otel-collector/otel-cm-duplicate-timeseries.yaml", "--ignore-not-found")
+		if t.Failed() {
+			nt.PodLogs("config-management-monitoring", ocmetrics.OtelCollectorName, "", false)
+		}
+	})
+
+	// If Workload Identity enabled on cluster, setup KSA to GSA annotation
+	if workloadPool, err := getWorkloadPool(nt); err != nil {
+		nt.T.Fatal(err)
+	} else if workloadPool != "" {
+		nt.T.Log("Workload identity enabled, adding KSA annotation to use %s service account", MonitorGSA)
+		nt.MustKubectl("annotate", "serviceaccount", "--namespace", ocmetrics.MonitoringNamespace, DefaultMonitorKSA,
+			fmt.Sprintf("iam.gke.io/gcp-service-account=%s@%s.iam.gserviceaccount.com", MonitorGSA, *e2e.GCPProject))
+	}
+
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap, log and IAM")
+	nomostest.DeletePodByLabel(nt, "app", ocmetrics.OpenTelemetry, false)
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := watchForOtelCollectorStatusCurrent(nt); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Watch for metrics in GCM, timeout 2 minutes")
+	ctx := context.Background()
+	client, err := createGCMClient(ctx)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	// retry for 2 minutes until metric is accessible from GCM
+	_, err = retry.Retry(120*time.Second, func() error {
+		return watchForMetricInGCM(ctx, nt, client, -2*time.Minute, fmt.Sprintf("%s/apply_operations_total", GCMMetricPrefix))
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Checking the otel-collector log contains no failure...")
+	err = CheckDeploymentLogHasNoFailure(nt, ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace, GCMExportErrorCaption)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// The ConfigMap that is expected to trigger duplicate time series error has
+	// name 'otel-collector-custom', which by the setup in otel-collector deployment
+	// will take precedence over 'otel-collector-googlecloud' that was deployed
+	// by the test.
+	nt.T.Log("Apply custom otel-collector ConfigMap that could cause duplicate time series error")
+	nt.MustKubectl("apply", "-f", "../testdata/otel-collector/otel-cm-duplicate-timeseries.yaml")
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap and log")
+	nomostest.DeletePodByLabel(nt, "app", ocmetrics.OpenTelemetry, false)
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := watchForOtelCollectorStatusCurrent(nt); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Checking the otel-collector log contains failure...")
+	_, err = retry.Retry(60*time.Second, func() error {
+		return CheckDeploymentLogHasFailure(nt, ocmetrics.OtelCollectorName, ocmetrics.MonitoringNamespace, GCMExportErrorCaption)
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func CheckDeploymentLogHasFailure(nt *nomostest.NT, deployment, namespace, errorString string) error {
```

Comment:
Suggest renaming each of these `Check*` functions to `Validate*` for consistency with similar functions

Reply:
Done!